pkg/api/types.go

```diff
@@ -1569,72 +1569,26 @@ func (k *KubernetesConfig) GetAddonScript(addonName string) string {
 	return kubeAddon.Data
 }
 
-// isAddonEnabled checks whether a k8s addon with name "addonName" is enabled or not based on the Enabled field of KubernetesAddon.
-// If the value of Enabled in nil, the "defaultValue" is returned.
-func (k *KubernetesConfig) isAddonEnabled(addonName string, defaultValue bool) bool {
+// IsAddonEnabled checks whether a k8s addon with name "addonName" is enabled or not based on the Enabled field of KubernetesAddon.
+// If the value of Enabled is nil, the "defaultValue" is returned.
+func (k *KubernetesConfig) IsAddonEnabled(addonName string) bool {
 	kubeAddon := k.GetAddonByName(addonName)
-	return kubeAddon.IsEnabled(defaultValue)
-}
-
-// IsMetricsServerEnabled checks if the metrics server addon is enabled
-func (o *OrchestratorProfile) IsMetricsServerEnabled() bool {
-	return o.KubernetesConfig.isAddonEnabled(MetricsServerAddonName,
-		common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.9.0"))
-}
-
-// IsAzureCNIMonitoringEnabled checks if the azure cni monitoring addon is enabled
-func (k *KubernetesConfig) IsAzureCNIMonitoringEnabled() bool {
-	return k.isAddonEnabled(AzureCNINetworkMonitoringAddonName, DefaultAzureCNIMonitoringAddonEnabled)
-}
-
-// IsContainerMonitoringEnabled checks if the container monitoring addon is enabled
-func (k *KubernetesConfig) IsContainerMonitoringEnabled() bool {
-	return k.isAddonEnabled(ContainerMonitoringAddonName, DefaultContainerMonitoringAddonEnabled)
-}
-
-// IsTillerEnabled checks if the tiller addon is enabled
-func (k *KubernetesConfig) IsTillerEnabled() bool {
-	return k.isAddonEnabled(TillerAddonName, DefaultTillerAddonEnabled)
+	return kubeAddon.IsEnabled()
 }
 
 // IsAADPodIdentityEnabled checks if the AAD pod identity addon is enabled
 func (k *KubernetesConfig) IsAADPodIdentityEnabled() bool {
```

Comment:
why are some of these addon-specific functions still needed? why not use `IsAddonEnabled(addonName)` everywhere?

Reply:
template layer stuff; using this example:

```
{{if .OrchestratorProfile.KubernetesConfig.IsAADPodIdentityEnabled}}
    "kubernetesAADPodIdentityEnabled": {
      "defaultValue": false,
      "metadata": {
        "description": "AAD Pod Identity status"
      },
      "type": "bool"
    },
{{end}}
```

Reply:
That can all probably be simplified in the future btw